graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+        requestBody:
+          $ref: '#/components/requestBodies/GraphQLRequest'
+        responses:
+          '200':
+            $ref: '#/components/responses/GraphQLSuccess'
+          '400':
+            $ref: '#/components/responses/GraphQLError'
+
+components:
+  securitySchemes:
+    BearerAuth:
+      description: Generic Bearer Authentication is supported using JWTs
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+    AwsBearerAuth:
+      description: |
+        For AWS usage we support JWTs that are injected into requests by the ELB.  
+        Since these come in a non-standard HTTP header we have to declare this as
+        an API Key scheme in OpenAPI spec as it doesn't have a notion of using a
+        HTTP scheme in a non-standard way (yay AWS weirdness!)
+      type: apiKey
+      in: header
+      name: X-Amzn-Oidc-Data
+  parameters:
+      dataset:
+        name: dataset
+        in: query
+        required: true
+        schema:
+          type: string
+        description: The underlying dataset (i.e. Graph) to apply requests against.
+      graphQLQuery:
+          name: query
+          in: query
+          required: true
+          schema:
+            type: string
+          description: The GraphQL query string.
+      sparqlQuery:
+          name: query
+          in: query
+          required: true
+          schema:
+            type: string
+          description: The SPARQL query string.
+      operationName:
+        name: operationName
+        in: query
+        required: false
+        schema:
+          type: string
+        description: Name of the operation to execute (optional).
+      variables:
+        name: variables
+        in: query
+        required: false
+        schema:
+          type: string
+          description: JSON-encoded variables for the query (optional).
+      extensions:
+        name: extensions
+        in: query
+        required: false
+        schema:
+          type: string
+        description: Extensions for the GraphQL request (optional).
+
+  requestBodies:
+      GraphQLRequest:
+        description: GraphQL request payload.
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                query:
+                  type: string
+                operationName:
+                  type: string
+                variables:
+                  type: object
+                extensions:
+                  type: object
+              required:
+                - query
+
+  responses:
+      GraphQLSuccess:
+        description: Successful response containing query results.
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                data:
+                  type: object
+      GraphQLError:
+        description: Error response for invalid GraphQL requests.
```

Comment:
GraphQL has a specific error response format defined at https://spec.graphql.org/draft/#sec-Response-Format as opposed to the generic Problem JSON format we use on other services

Reply:
Again, the Resources in GraphQL Jena  suggested otherwise. I'll update it